service/src/main/java/uk/gov/hmcts/reform/fpl/config/UserEmailLookupConfiguration.java

```diff
@@ -0,0 +1,35 @@
+package uk.gov.hmcts.reform.fpl.config;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+@Configuration
+public class UserEmailLookupConfiguration {
+
+    private static final String MAPPINGS_SEPARATOR = ";";
+    private static final String LA_SEPARATOR = "=>";
+    private static final String USER_SEPARATOR = ",";
+    private final Map<String, List<String>> mapping;
+
+    public UserEmailLookupConfiguration(@Value("${fpl.hmcts_email.mapping}") String config) {
+        ImmutableMap.Builder<String, List<String>> emails = ImmutableMap.builder();
+
+        Arrays.stream(config.split(MAPPINGS_SEPARATOR)).forEach(entry -> {
```

Comment:
This is n-th time we do it. We could try to extract logic into reusable methods. We always gonna have either `String=>String` mapping or `String=>String[]` mappings.